variables.tf

```diff
@@ -57,6 +57,25 @@ variable "agents_min_count" {
   description = "Minimum number of nodes in a pool"
 }
 
+variable "http_proxy_config" {
+  type = object({
+    http_proxy  = optional(string)
+    https_proxy = optional(string)
+    no_proxy    = optional(list(string))
+    trusted_ca  = optional(string)
+  })
+  default     = {}
+  description = <<-EOT
+    optional(object({
+      http_proxy  = (Optional) The proxy address to be used when communicating over HTTP. Changing this forces a new resource to be created.
+      https_proxy = (Optional) The proxy address to be used when communicating over HTTPS. Changing this forces a new resource to be created.
+      no_proxy    = (Optional) The list of domains that will not use the proxy for communication. Note: If you specify the `default_node_pool.0.vnet_subnet_id`, be sure to include the Subnet CIDR in the `no_proxy` list. Note: You may wish to use Terraform's `ignore_changes` functionality to ignore the changes to this field.
```

Comment:
There's no way to config module's resource's `ignore_changes` list dynamically. I'd like to add this `no_proxy` into `ignore_changes` by the following steps:

1. Add the attribute to `azurerm_kubernetes_cluster`'s `ignore_changes`:
```hcl
ignore_changes = [
  kubernetes_version,
  http_proxy_config[0].no_proxy
]
```
2. Add a corresponding `null_resource` to recreate the aks cluster when user do changed `no_proxy`:
```hcl
resource "null_resource" "aks_cluster_recreate" {
  triggers = {
    http_proxy_no_proxy = try(join(",", var.http_proxy_config.no_proxy), "")
  }
}
```
3. Add `replace_triggered_by` to `azurerm_kubernetes_cluster` resource:
```hcl
replace_triggered_by = [null_resource.aks_cluster_recreate]
```

I've tried to add toggle `count` to this new `null_resource.aks_cluster_recreate` resource but we cannot use this resource with `count = 0` along with `replace_triggered_by`, so we must add this `null_resource` resource, which makes this pr a breaking change. We have to merge this pr in our next major version upgrade, which I hope could be finished in one month.

Reply:
Fixed, thanks for the suggestion!